### PR TITLE
 persist consent to key submission (EXPOSUREAPP-3734)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Creates new SubmissionSettings class which persists if the user has given consent for the submission of keys

### CODE REVIEW

**Review by ralfgehrer — COMMENTED** [review]
Some tests would be nice :)

**Comment by vaubaehn** [comment]
Hi @chiljamgossow , welcome to our small universe, nice to have you onboard,
and
hi @ralfgehrer ,

may I ask shortly about your PR - I bet there will be questions on our slack channel...

What's the rationale of persisting user consent for TEK submission?
Is it just to have/use information about that user gave consent (e.g., logging/testing)?
Or is there any planning towards future automatic submission of TEKs? Afaik, for now ENF would require consent once every 24 hours - will there be a change from Google/Apple in future?

Kind regards, v.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@Singleton
+class SubmissionSettings @Inject constructor(
+    @AppContext private val context: Context
+) {
+
+    private val prefs by lazy {
+        context.getSharedPreferences("submission_localdata", Context.MODE_PRIVATE)
+    }
+
+    var hasGivenConsent = FlowPreference(
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt:18]: Can this be `val`?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        preferences = prefs,
+        key = "key_submission_consent",
+        reader = FlowPreference.basicReader(false),
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt:21]: Try using `prefs.createFlowPreference(key = "key_submission_consent", defaultValue = false)`

**Comment by chiljamgossow** [comment]
Needed to move to a new PR as this one does not run the checks

**Comment by chiljamgossow** [comment]
see new PR https://github.com/corona-warn-app/cwa-app-android/pull/1633

> Hi @chiljamgossow , welcome to our small universe, nice to have you onboard,
> and
> hi @ralfgehrer ,
> 
> may I ask shortly about your PR - I bet there will be questions on our slack channel...
> 
> What's the rationale of persisting user consent for TEK submission?
> Is it just to have/use information about that user gave consent (e.g., logging/testing)?
> Or is there any planning towards future automatic submission of TEKs? Afaik, for now ENF would require consent once every 24 hours - will there be a change from Google/Apple in future?
> 
> Kind regards, v.

Hi @vaubaehn, we are persisting the consent to be able to display to the user if consent has been given on a new screen. This is part of a redesign of the submission flow. There are no automatic submissions planned. Kind regards.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt:18]: yes, changed it

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt:21]: done

**Comment by vaubaehn** [comment]
@chiljamgossow Thanks for your reply! Sounds like a good idea :+1: In case any questions arise on user side, I now can refer correct information. Have a good day!

**Pull request closed without merging**